rules/java/shared/lang/user_input.yml

```diff
@@ -39,6 +39,21 @@ patterns:
       - variable: JAVA_SHARED_LANG_USER_INPUT_PARAM_MAP
         detection: java_shared_lang_user_input_param_map
         scope: cursor
+  - pattern: $<JAVA_SHARED_LANG_USER_INPUT_PARAM_MAP>.get()[$<_>];
+    filters:
+      - variable: JAVA_SHARED_LANG_USER_INPUT_PARAM_MAP
+        detection: java_shared_lang_user_input_param_map
+        scope: cursor
+  - pattern: $<JAVA_SHARED_LANG_USER_INPUT_PARAM_NAMES>.nextElement();
+    filters:
+      - variable: JAVA_SHARED_LANG_USER_INPUT_PARAM_NAMES
+        detection: java_shared_lang_user_input_param_names
+        scope: cursor
+  - pattern: $<JAVA_SHARED_LANG_USER_INPUT_PARAM_VALUES>[$<_>];
+    filters:
+      - variable: JAVA_SHARED_LANG_USER_INPUT_PARAM_VALUES
+        detection: java_shared_lang_user_input_param_values
+        scope: cursor
```

Comment:
Maybe there is a smarter way to do this. My aim is to catch weird access cases when param values are e.g. an array: 

```java
    String[] parameterValues = request.getParameterValues("URL");
    String indexedParameterValue = parameterValues[0];

    Enumeration<String> parameterNames = request.getParameterNames();
    String parameterNamesElem = parameterNames.nextElement();

    Map<String, String[]> parameterMap = request.getParameterMap();
    String indexedValueFromParameterMap = parameterMap.get("URL")[0];
```